### ISSUE TITLE
3.0.0 Seems to break the simple http server

### ISSUE THREAD

**Issue by No9** [feross/chrome-net#25]
Understand the PR was taken and version bumped  with "an abundance of caution" :) 
But when this module is now used with http-node the basic node http server is broken 
Was working with chrome-net@2.5 http-node has not had a revision
```
var http = require('http');
http.createServer(function (req, res) {
  res.writeHead(200, {'Content-Type': 'text/plain'});
  res.end('Hello World\n');
}).listen(1337, '127.0.0.1');
console.log('Server running at http://127.0.0.1:1337/');
```
I have put together a full example here with instructions to reproduce
https://github.com/No9/simple-chrome-http-server 

The request does seem to be received by the server and a response is generated by the HTTP layer 
but it never seems to make it back down to the net layer. 
Hoping @jscissr might see something obvious on this. 

Happy to add this as a formal test but didn't know if it should go here or in node-http?

**Comment by jscissr** [comment]
Could you try what happens when you put the server into the window page instead of the background page? The background page has some special quirks like it can be suspended. I think it *should* work, though.

**Comment by No9** [comment]
Hey @jscissr  I seem to remember seeing funny suspended behaviour when there was no window at all but not when it was in a background page that had a front page. 
Anyhoo I created a branch with the server in the index.html but it has exactly the same issue.
Call just hangs and eventually times out with a "no data returned" message 
Branch is here FYI https://github.com/No9/simple-chrome-http-server/tree/server-in-index

**Comment by feross** [comment]
I'm sorry - I don't use this module anymore and don't have time to look into this.

@jscissr - You're a collaborator now. Can you fix this up?

**Comment by jscissr** [comment]
I'll look into it later today. I could reproduce the problem.

**Comment by jscissr** [comment]
@feross Can you publish to npm, please?

**Comment by No9** [comment]
Sweet thanks @jscissr for turning this around so quick 
LGTM :+1:

**Comment by feross** [comment]
Published as 3.0.1.

**Comment by jscissr** [comment]
Thanks!
